### PR TITLE
initial commit

### PR DESCRIPTION
**Pull request by atorgayev**

### CODE REVIEW

File: options/preprocess.proto [hunk 1]
```diff
@@ -0,0 +1,19 @@
+syntax = "proto2";
```
Comment (Evgeniy-L) [inline options/preprocess.proto]: Please change to `syntax = "proto3"` as we try to stick to proto3 syntax for infoblox plugins. I also see that you use `proto3` for the demo.

File: middleware/preprocessor.go [hunk 2]
```diff
@@ -0,0 +1,26 @@
+package grpc_preprocessor
```
Comment (Evgeniy-L) [inline middleware/preprocessor.go]: I do not really like the package name here. I think `package middleware` will look more consistent here.

File: README.md [hunk 3]
```diff
@@ -1 +1 @@
-# protoc-gen-preprocess
\ No newline at end of file
+# protoc-gen-preprocess
```
Comment (Evgeniy-L) [inline README.md]: Please add some details in readme about what this plugins does and how to use it.

File: options/preprocess.proto (hunk 1 above)
Comment (atorgayev) [reply options/preprocess.proto]: ok

File: middleware/preprocessor.go (hunk 2 above)
Comment (atorgayev) [reply middleware/preprocessor.go]: ok

**Comment by dmacthedestroyer** [comment]
what is this?

**Comment by dmacthedestroyer** [comment]
oohh a "Pull Request"! I'll Google it later. "Automatic data pre-processing" is still a pretty vague term -- I guess I'll stay tuned to see what comes of this.

File: README.md (hunk 3 above)
Comment (atorgayev) [reply README.md]: Added readme and example application

**Pull request merged**